ca/certificate-authority.go

```diff
@@ -325,10 +344,20 @@ func (ca *CertificateAuthorityImpl) IssueCertificate(csr x509.CertificateRequest
 	serialBigInt := big.NewInt(0)
 	serialBigInt = serialBigInt.SetBytes(serialBytes)
 
+	var profile string
+	switch key.(type) {
+	case *rsa.PublicKey, rsa.PublicKey:
```

Comment:
Have you found that IssueCertificate gets passed both pointer and non-pointer keys? I believe we should only be passing `*rsa.PublicKey` and `*ecdsa.PublicKey` everywhere, otherwise we run into problems with `crypto/x509`, which expects only the pointer version.

Assuming no mitigating factors, I'd prefer to see this take just the pointer versions, and return an error `fmt.Errorf("Invalid key type %t", key)` if there's not a match. My reasoning is that, even though we believe this should be unreachable, it's not such a critical error that we need to panic, which would have the potential to put the CA in a crash loop.